### PR TITLE
Added Coin Tower Problem Using Dynamic Programming, ISSUE #4506

### PR DESCRIPTION
**Pull request by SarthakChaudhary46**
Added Coin Tower Problem Using Dynamic Programming, ISSUE #4506

### CODE REVIEW

File: src/main/java/com/thealgorithms/dynamicprogramming/CoinTowerProblem.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    // Driver Program
+    public static void main(String[] args) {
+        int N = 5;
+        int X = 3;
+        int Y = 4;
+
+        char result = coinTowerWinner(N, X, Y);
+        System.out.println("The winner of the Coin Tower Game is Player " + result + ".");
+    }
+
```
Comment (vil02) [inline src/main/java/com/thealgorithms/dynamicprogramming/CoinTowerProblem.java:22]:
```suggestion
```

Please add proper tests into [`src/test/java/com/thealgorithms/dynamicprogramming`](https://github.com/TheAlgorithms/Java/tree/master/src/test/java/com/thealgorithms/dynamicprogramming).

Comment (vil02) [inline src/main/java/com/thealgorithms/dynamicprogramming/CoinTowerProblem.java:22]:
```suggestion
```

Please add proper tests into [`src/test/java/com/thealgorithms/dynamicprogramming`](https://github.com/TheAlgorithms/Java/tree/master/src/test/java/com/thealgorithms/dynamicprogramming).

**Review by vil02 — CHANGES_REQUESTED** [review]

**Comment by BamaCharanChhandogi** [comment]
I think this type of problem falls under algorithmic questions in the context of Data Structures and Algorithms (DSA).

**Comment by siriak** [comment]
It is an algorithmic question, but not a generalized algorithm. It's an application of dynamic programming to a specific problem. As such, we do not add specific implementations because there's an infinite number of them and they are all very similar. (To be completely fair, we add some to demonstrate the approach, but there are already enough examples in `dynamicprogramming` module).

**Comment by SarthakChaudhary46** [comment]
> @BamaCharanChhandogi, @siriak: having in mind this:
> 
> https://github.com/TheAlgorithms/Java/blob/a216cb8a59ad04b3cadcc156e69dea41f3d1b465/CONTRIBUTING.md?plain=1#L3
> 
> 
> do you think that this pull-requests solves a problem, which is _general enough_?

-I can either add different test cases as suggested by @vil02, or I can make it a general solution by taking custom inputs. Which option should I go with? For reference, I used src/main/java/com/thealgorithms/dynamicprogramming/ClimbingStairs.java, which is a general solution to a LeetCode problem, similar to what I did. @vil02

**Comment by siriak** [comment]
Is there a description of this problem somewhere other than on coding practice websites? Maybe Wikipedia or a book? If so, we can accept it as a popular enough. If no, it's just another problem that has slight variations in the problem statement, but doesn't warrant being included as a different algorithm (again, it's not the algorithm itself, but the dynamic programming approach that we are interested in). As you have mentioned, there are already other examples of dynamic programming in the repository that are very similar to this one.

**Comment by vil02** [comment]
@SarthakChaudhary46 I think we do not need another example demonstrating _dynamic programming_. Unless this is some well-know problem. If yes - could you provide some references?

**Comment by SarthakChaudhary46** [comment]
@vil02 @siriak I have generalized the problem; is it sufficiently comprehensive?

**Comment by vil02** [comment]
> @vil02 @siriak I have generalized the problem; is it sufficiently comprehensive?

No.

I guess you made some mistake in your last commit - I do not see any generalization and the tests are gone. Please check if this was some accident.

**Comment by siriak** [comment]
@vil02 but can it even be generalized? I don't think so. The problem itself is very specific, and I don't think we can help with that. There are already enough examples of dynamic programming in the repository, so I suggest just closing this PR.

**Comment by vil02** [comment]
@SarthakChaudhary46 as discussed earlier, this does not fit into this repository.

If you would be looking for an idea for a contribution, please let me know because I have few.

**Pull request closed without merging**